src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalTestDetectors.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Infrastructure.Common
+{
+    // This class contains helper "detector" methods to detect specific
+    // conditions about the environment in which the tests are run.
+    // They are used for [ConditionalFact] and [ConditionalTheory]
+    // tests.  They are called at most once, and the return value cached.
+    internal static class ConditionalTestDetectors
+    {
+        public static bool IsRootCertificateInstalled()
+        {
+            try
+            {
+                ServiceUtilHelper.EnsureRootCertificateInstalled();
+                return true;
```

Comment:
I'm not sure I totally agree but let's discuss.  If EnsureRootCertificateInstalled() said it succeeded, you have to trust it -- that's its job.  Or add code to it to double check (though I'm not sure what additional work it would do.  I don't think the detectors should second guess when EnsureRootCertificateInstalled said it succeeded.  The only place where it might make sense is in the exception path.  Maybe EnsureRootCertificateInstalled failed to write but there is a usable cert already there.  But even in that case, I believe that job belongs to EnsureRootCertificateInstalled, not the detector.